app/main.py

```diff
@@ -1,6 +1,55 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    """
+    decorating only functions that take immutable arguments.
+    cache should print "Getting from cache" when returns stored value
+    and "Calculating new result" when run function with new arguments.
+
+    structure for DB_cache:
+        db_cache =
+        {   "function's name1":
+                {   (tuple1 of arguments): result1,
+                    (tuple2 of arguments): result2,
+                    ...
+                    (tupleN of arguments): resultN,
+                }
+            "function's name2": {...}
+            "function's nameN": {...}
+        }
+    """
+    # 0. initialise DB_cache
+    db_cache = {}
```

Comment:
why names is db_cache? I don't see that description of task includes something about a db

Reply:
cache - decorator's name, so I need another name for variable, db - for me is nice prefix to something big. ok, I change